packages/react/components/overlay/src/Overlay.tsx

```diff
@@ -1,15 +1,42 @@
-import { Portal } from '@project44-manifest/react-portal';
+import * as React from 'react';
+import { Overlay as ReactAriaOverlay } from '@react-aria/overlays';
 import { Fade } from '@project44-manifest/react-transition';
 import type { OverlayProps } from './Overlay.types';
 
 export function Overlay(props: OverlayProps) {
-  const { children, containerRef, isOpen } = props;
+  const { children, containerRef, isOpen, onEntered, onExited } = props;
 
-  const contents = (
-    <Fade appear unmountOnExit in={isOpen}>
-      <div>{children}</div>
-    </Fade>
+  const [exited, setExited] = React.useState(!isOpen);
+
+  const handleEntered = React.useCallback(
+    (node: HTMLElement, isAppearing: boolean) => {
+      setExited(false);
+
+      onEntered?.(node, isAppearing);
+    },
+    [onEntered],
+  );
+
+  const handleExited = React.useCallback(
+    (node: HTMLElement) => {
+      setExited(true);
+
+      onExited?.(node);
+    },
+    [onExited],
   );
 
-  return <Portal containerRef={containerRef}>{contents}</Portal>;
+  const mountOverlay = isOpen || !exited;
+
+  if (!mountOverlay) {
+    return null;
+  }
+
+  return (
+    <ReactAriaOverlay portalContainer={containerRef?.current as Element}>
```

Comment:
Using the react-aria overlay element to ensure focus scope is handled correctly.